### ISSUE TITLE
Unexpected keyword arguments in constructor call for inherited dataclass

### ISSUE THREAD

**Issue by dwc-pixel** [PyCQA/pylint#3201]
Hello.

When handling dataclasses, if there is an init method in parent class, pylint sees this as an exception for child class arguments. The code runs without an exception but pylint complains as errors.

### Steps to reproduce
1.Install python 3.7
2.Install pylint (2.3.1)
3.Run code

<pre>
$ cat dataclass_test.py
import dataclasses

@dataclasses.dataclass
class MyBaseDataClass:
  arg0: str = None

  def __init__(self, arg0):
    raise NotImplementedError

@dataclasses.dataclass
class MyDataClass(MyBaseDataClass):
  arg1: float = None
  arg2: str = None

if __name__ == '__main__':
  try:
    my_base_data_class = MyBaseDataClass(arg0='base')
    print(my_base_data_class)
  except NotImplementedError:
    print('expected')

  my_data_class = MyDataClass(arg1=0, arg2='a')
  print(my_data_class)
$ pylint -E dataclass_test.py
************* Module dataclass_test
dataclass_test.py:22:18: E1123: Unexpected keyword argument 'arg1' in constructor call (unexpected-keyword-arg)
dataclass_test.py:22:18: E1123: Unexpected keyword argument 'arg2' in constructor call (unexpected-keyword-arg)
dataclass_test.py:22:18: E1120: No value for argument 'arg0' in constructor call (no-value-for-parameter)
$
</pre>

### Expected behavior
No error reported

### pylint --version output
2.3.1

**Comment by AWhetter** [comment]
I cannot reproduce on the latest master. Could you please confirm?

**Comment by dwc-pixel** [comment]
I'm sorry for the trouble. I must've done something weird when doing the copy paste. I pasted the code again.

**Comment by dwc-pixel** [comment]
@AWhetter Didn't realize comment was a mark down. All indentations were removed. I fixed the comment on steps to reproduce. Thank you for looking into this.

**Comment by roytseng-tw** [comment]
I can reproduce the same error with
```
pylint 2.5.3
astroid 2.4.2
Python 3.7.6 
```

**Comment by Db-pckr** [comment]
I just found similar error in my code with:

pylint 2.6.0
astroid 2.4.2
Python 3.8.5

**Comment by thefifo** [comment]
+1

pylint 2.7.4
astroid 2.5.3
Python 3.7.3
